core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -4341,18 +4341,13 @@ private boolean isValidTemporalType(Type type)
 
         private PointerType toPointerType(QueryPeriod.RangeType type)
         {
-            PointerType pointerType = null;
             switch (type) {
                 case TIMESTAMP:
-                    pointerType = PointerType.TEMPORAL;
-                    break;
+                    return PointerType.TEMPORAL;
                 case VERSION:
-                    pointerType = PointerType.TARGET_ID;
-                    break;
-                default:
```

Comment:
Why? The previous code was more readable, IMO. The switch was “exhaustive”. Now I need to look past the switch to see what happens after.

Reply:
The new version allows for a static analysis (error-prone) to verify the switch is exhaustive, so a human doesn't need to do that.

From human perspective, the code is equally clear -- whether the switch ends with `default: throw;` or `} throw` does put the default handling (`throw`) further from the reader.

Reply:
How is error prone able to prove its exhaustive, since it’s now missing a default case?

Reply:
it sees that it covers all options defined in the enum.

it can verify that because errorprone is a compiler, so it knows all things a compiler would know